packages/loot-core/src/client/queries/queriesSlice.ts

```diff
@@ -0,0 +1,867 @@
+  lastTransaction: TransactionEntity | null;
+  updatedAccounts: Array<AccountEntity['id']>;
+  accounts: AccountEntity[];
+  accountsLoaded: boolean;
+  categories: CategoryViews;
+  categoriesLoaded: boolean;
+  commonPayeesLoaded: boolean;
+  commonPayees: PayeeEntity[];
+  payees: PayeeEntity[];
+  payeesLoaded: boolean;
+};
+
+const initialState: QueriesState = {
+  newTransactions: [],
+  matchedTransactions: [],
+  lastTransaction: null,
+  updatedAccounts: [],
+  accounts: [],
+  accountsLoaded: false,
+  categories: {
+    grouped: [],
+    list: [],
+  },
+  categoriesLoaded: false,
+  commonPayees: [],
+  commonPayeesLoaded: false,
+  payees: [],
+  payeesLoaded: false,
+};
+
+type MarkAccountReadPayload = {
+  id: AccountEntity['id'];
+};
+
+// Account actions
+
+type CreateAccountPayload = {
+  name: AccountEntity['name'];
+  balance: AccountEntity['balance_current'];
+  offBudget: boolean;
+};
+
+export const createAccount = createAppAsyncThunk(
+  `${sliceName}/createAccount`,
+  async ({ name, balance, offBudget }: CreateAccountPayload, { dispatch }) => {
+    const id: AccountEntity['id'] = await send('account-create', {
+      name,
+      balance,
+      offBudget,
+    });
+    await dispatch(getAccounts());
+    await dispatch(getPayees());
+    return id;
+  },
+);
+
+type CloseAccountPayload = {
+  id: AccountEntity['id'];
+  transferAccountId?: AccountEntity['id'];
+  categoryId?: CategoryEntity['id'];
+  forced?: boolean;
+};
+
+export const closeAccount = createAppAsyncThunk(
+  `${sliceName}/closeAccount`,
+  async (
+    { id, transferAccountId, categoryId, forced }: CloseAccountPayload,
+    { dispatch },
+  ) => {
+    await send('account-close', {
+      id,
+      transferAccountId: transferAccountId || null,
+      categoryId: categoryId || null,
+      forced,
+    });
+    dispatch(getAccounts());
+  },
+);
+
+type ReopenAccountPayload = {
+  id: AccountEntity['id'];
+};
+
+export const reopenAccount = createAppAsyncThunk(
+  `${sliceName}/reopenAccount`,
+  async ({ id }: ReopenAccountPayload, { dispatch }) => {
+    await send('account-reopen', { id });
+    dispatch(getAccounts());
+  },
+);
+
+type UpdateAccountPayload = {
+  account: AccountEntity;
+};
+
+export const updateAccount = createAppAsyncThunk(
+  `${sliceName}/updateAccount`,
+  async ({ account }: UpdateAccountPayload) => {
+    await send('account-update', account);
+    return account;
+  },
+);
+
+export const getAccounts = createAppAsyncThunk(
+  `${sliceName}/getAccounts`,
+  async () => {
+    const accounts: AccountEntity[] = await send('accounts-get');
+    return accounts;
+  },
+);
+
+// Category actions
+
+type CreateGroupPayload = {
+  name: CategoryGroupEntity['name'];
+};
+
+export const createGroup = createAppAsyncThunk(
+  `${sliceName}/createGroup`,
+  async ({ name }: CreateGroupPayload, { dispatch }) => {
+    const id = await send('category-group-create', { name });
+    dispatch(getCategories());
```

Comment:
I think this should be `await`ed (and maybe also `.unwrap`ed)

Reply:
I think our general guideline for dispatching thunks would be:
- Do we need to use the thunk result in downstream logic?
  - If yes, we should `await` the thunk to ensure we get latest state/data to use in the downstream logic
  - If not, we should just `dispatch` it without `await` and let the components react to the state change asynchronously